src/main/java/ekud/ui/Ui.java

```diff
@@ -0,0 +1,86 @@
+package ekud.ui;
+
+import java.util.List;
+
+import ekud.command.Command;
+import ekud.error.EkudException;
+import ekud.state.Task;
+import ekud.state.TaskList;
+
+/**
+ * Represents any user interface to ekud.
+ * This could be in the form of a command-line interface or a graphical user
+ * interface.
+ */
+public abstract class Ui {
+    private CommandHandler handler;
+
+    public void setOnCommand(CommandHandler handler) {
+        this.handler = handler;
+    }
+
+    protected boolean handle(Command command) {
+        if (handler == null) {
+            return true;
+        }
+        return handler.handle(command);
+    }
```

Comment:
Here is another boolean method whose name may need to be renamed slightly 😄

Reply:
Refer to https://github.com/nus-cs2103-AY2324S1/ip/pull/163#discussion_r1318816451